src/Raven.Embedded/EmbeddedServer.cs

```diff
@@ -71,9 +71,16 @@ public async Task RestartServerAsync()
             await StartServerInternalAsync().ConfigureAwait(false);
         }
 
-        public void StartServer(ServerOptions? options = null)
+
+        private CancellationToken _shutdownToken;
+        private CancellationTokenRegistration _shutdownRegistration;
+
+        // Solution 1: Pass a shutdown token to start
+        public void StartServer(ServerOptions? options = null, CancellationToken shutdownToken)
         {
             _serverOptions = options ??= ServerOptions.Default;
+            _shutdownToken = shutdownToken;
+            _shutdownRegistration = shutdownToken.Register(()=>{this.Dispose()})
```

Comment:
That is not something that you _have_ to do in RavenDB code, though, right?
You can call that from your own code